src/firefly/js/api/ApiUtil.js

```diff
@@ -94,54 +93,33 @@ export function unrenderDOM(div) {
 
 /**
  * @summary Add a listener to any action type
- * @param {string} actionType a string or and array of strings. Each string is an action constant from firefly.action.type
- * @param {function} callBack the call back will be call with two parameters: action object and state object
+ * @param {string} actionType a string or an array of strings. Each string is an action constant from firefly.action.type
+ * @param {function} callBack the call back will be called with three parameters: action object, state object, and extraData
  *                 If it returns true the listener will be removed.
- * @param {object} extraData, an object with data to send to the callback, can be anything
+ * @param {object} extraData an object with data to send to the callback, can be anything
  *
  * @return {function} a function that will remove the listener
  * @public
  * @func addActionListener
  * @memberof firefly.util
  */
 export function addActionListener(actionType,callBack, extraData) {
```

Comment:
I wonder if this is needed anymore.  This was created to shield API users from saga.  dispatchAddActionWatcher is doing the same.  Maybe it should be deprecated?

Reply:
We might want to keep it for backward compatibility and simplicity. We don't have to explain flux dispatch and getStore parameters. We did not add saga or watcher dispatchers to the javadoc, this is our internal implementation of action listeners, that we advertise in public API.

Reply:
I am pretty sure IRSA is already using this.  We should keep it at least until they give us suggestions (Jeff is planning to) about how we could improve it.